### DISCUSSION TITLE
hybrid search and vector fusion

### DISCUSSION THREAD

**Discussion by branaway** [milvus-io/milvus#34722] [Q&A and General discussion]
How do I know if a hybrid search uses vector fusion or not? Weird weird in the entire circuit code to our specified that the vector fusion be used for hybrid searching? It appears he might use cases that Victor fusion is the way to go for better performance and four the overall relevance of the results to make query. So please can someone explain to me the relationship of vecctor fusion to the hybrid searching? Thanks

**Comment by xiaofan-luan** [comment]
We don't combine different vectors into one. I know some other database is doing so, but this is definitely a wrong thing.

Milvus do separate on different columns and use a ranking to combine search result from different columns. that is the hybrid search we defined.

**Reply by branaway** [reply]
but the 2021 paper mentioned vector fusion a lot and showed that it performed very favorably. What has happened? It's been dropped since?

Back the hybrid searching API. 1) does the order of the queries on vectors matter? 2) why does one need to  specify a K for each query on vectors AND the reranking function?

**Reply by xiaofan-luan** [reply]
1） it doesn't, both search happened in parallel
2)  it does offer flexibility, some times people need to find more result on retrieve more and rerank

**Reply by branaway** [reply]
Thanks for the API clarification.  Let me try to understand more how it works.  assume I have two vectors to search on. These two vectors(embeddings) describe different aspects of the same entry. Example I'm trying to work on a dating website that helps people to meet their dates' requirement.  one dimension would be the age of the other dimension will be the look. I have used very low-dimensional vector to represent the age range and look range. These two aspects of people are orthogonal. a man is looking for someone around mid 30s with decent look. As you can see the condition is not or can hardly be precisely described using scalars. And also the user put more weight on the look than the age. Now separately Milvus searches on the age and look vectors and come back with your own list of candidates.  When the age match comes back it has candidates with perfect age but it may have really really bad look.  Similarly the look search  comes back with a list of very beautiful candidates but all of them are in early 20s. Now when you merge them together you end up with a list of beautiful people in the early 20s, because you have put the weight on the look,  so the final list is overwhelmingly taken by a good look candidates with ages falling out of the range. And in extreme cases the intersection of the two results that can be empty. 

So my question is how Milvus would reconcile this conflict situations. How should I as the developer  give Milvus better cues for more reasonable results. 

Thanks

**Reply by xiaofan-luan** [reply]
Is there a chance to put age and look into one embedding, I think it's pretty easy for language model to do so.

on the other side, you can combine two vector into one by your self, you can do:
normalize(vectorA) + 2 * normalize(vectorB) and build one index on it

**Reply by branaway** [reply]
But then how to factor in the weights that are user specific? One favors look over age and another person goes another way around. I guess I  can do the vector concentration on the query side only. Is this actually what the 'vector fusion' about?
